File: python/ReverseLinkedList.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def reverseList(self, head):
        if head is None or head.next is None:
            return head
        curr = head
        prev = None
        while curr is not None:
            next = curr.next
            curr.next = prev
            prev = curr
            curr = next
        return prev

File: js/ReverseLinkedList.js

function ListNode(val){
    this.val = val;
    this.next = null;
}

var reverseList = function(head){
    if(head === null || head.next === null){
        return head;
    } else {
        var prev = null;
        var curr = head;
        var next = curr.next;
        while(curr !== null){
            next = curr.next;
            curr.next = prev;
            prev = curr;
            curr = next;
        }
        return prev;
    }
};